### ISSUE TITLE
Hidive - For videos with dubs -> audio language "random"

### ISSUE THREAD

**Issue by nachfuellbar** [yt-dlp/yt-dlp#4544]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.07.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Germany

### Provide a description that is worded well enough to be understood

When downloading videos from Hidive i get a random audio language for videos with dubs available.
For example "The Demon Girl Next Door" Season 1 Episode 1 is always japanse which is okay, at least for me.
BUT Episode 2 will ALWAYS downloaded with english audio - i tried some with adding options like --video-multistreams, audio-multistreams and --extractor-args "hidive:language=ja" and similar but i always get the same audio language.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['--all-subs', '-n', '--batch-file', '/home/XXX/hidive/The_Demon_Girl_Next_Door', '-vU']
[debug] User config "/home/XXX/.config/yt-dlp/config": ['--cookies-from-browser', 'chrome', '--video-multistreams', '--audio-multistreams', '--all-subs', '--abort-on-unavailable-fragment', '--abort-on-error', '-o', 'XXX/%(title)s-%(id)s%(language)s.%(ext)s', '--no-color']
[debug] Batch file urls: ['https://www.hidive.com/stream/the-demon-girl-next-door/s01e002']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.07.18 [135f05e]
[debug] Python 3.10.5 (CPython 64bit) - Linux-5.18.15-zen1-2-zen-x86_64-with-glibc2.36 (glibc 2.36)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.12.0, brotlicffi-1.0.9.2, certifi-2022.06.15, secretstorage-3.3.2, sqlite3-2.6.0
[Cookies] Extracting cookies from chrome
[debug] Extracting cookies from: "/home/XXX/.config/google-chrome/Default/Cookies"
[debug] detected desktop environment: GNOME
[debug] Chosen keyring: GNOMEKEYRING
[Cookies] Extracted 431 cookies from chrome
[debug] cookie version breakdown: {'v10': 0, 'v11': 425, 'other': 0, 'unencrypted': 6}
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.07.18, Current version: 2022.07.18
yt-dlp is up to date (2022.07.18)
[HiDive] None: Downloading webpage
[HiDive] Logging in
[debug] [HiDive] Extracting URL: https://www.hidive.com/stream/the-demon-girl-next-door/s01e002
[HiDive] the-demon-girl-next-door/s01e002: Downloading JSON metadata
[HiDive] the-demon-girl-next-door/s01e002: Downloading m3u8 information
[HiDive] the-demon-girl-next-door/s01e002: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[HiDive] the-demon-girl-next-door/s01e002: Downloading webpage
[HiDive] the-demon-girl-next-door/s01e002: Downloading xx subtitle information
[HiDive] the-demon-girl-next-door/s01e002: Downloading en subtitle information
[debug] Downloading subtitles: english-subs, english-caps
[debug] Default format spec: bestvideo+bestaudio/best
[info] the-demon-girl-next-door/s01e002: Downloading 1 format(s): en_hv_or-14524
Deleting existing file XXX/the-demon-girl-next-door_s01e002-the-demon-girl-next-door_s01e002en.english-subs.vtt
[info] Writing video subtitles to: XXX/the-demon-girl-next-door_s01e002-the-demon-girl-next-door_s01e002en.english-subs.vtt
[debug] Invoking http downloader on "https://www.hidive.com/caption/vtt/6fe90974bb5d573d5bd5de7b2e777f19dc99c9f4/MMZ_s01e002_tv_hv_or_ja_en_v00.vtt"
[download] Destination: XXX/the-demon-girl-next-door_s01e002-the-demon-girl-next-door_s01e002en.english-subs.vtt
[download] 100% of 65.62KiB in 00:01
Deleting existing file XXX/the-demon-girl-next-door_s01e002-the-demon-girl-next-door_s01e002en.english-caps.vtt
[info] Writing video subtitles to: XXX/the-demon-girl-next-door_s01e002-the-demon-girl-next-door_s01e002en.english-caps.vtt
[debug] Invoking http downloader on "https://www.hidive.com/caption/vtt/6fe90974bb5d573d5bd5de7b2e777f19dc99c9f4/MMZ_s01e002_tv_hv_or_en_en_v01.vtt"
[download] Destination: XXX/the-demon-girl-next-door_s01e002-the-demon-girl-next-door_s01e002en.english-caps.vtt
[download] 100% of 11.97KiB in 00:01
[debug] Invoking hlsnative downloader on "https://www.hidive.com/manifest/child/6fe90974bb5d573d5bd5de7b2e777f19dc99c9f4/MMZ_s01e002_tv_hv_or_en_xx_HLS1080p_16x9_00_v.m3u8?#="
[download] XXX/the-demon-girl-next-door_s01e002-the-demon-girl-next-door_s01e002en.mp4 has already been downloaded
[download] 100% of 340.74MiB
```

**Comment by pukkandan** [comment]
https://github.com/yt-dlp/yt-dlp#format-selection

**Comment by CZeil** [comment]
Try this
`-f "(b*[height<=1080][language*=ja])"`

**Comment by CZeil** [comment]
Your problem (as I see it) is that you are not specifying that you want Japanese audio, so hidive downloads what it wants. You put a --extractor-args but hidive does not have --extractor-args in the code.

**Comment by nachfuellbar** [comment]
the -f variant works
would be great if there are more examples for something like this in the readme

but I don't have an real issue anymore

**Comment by CZeil** [comment]
In fact, if there are such examples, what you need to do is to experiment a bit with the filtering options. That's what I used to do in my early days in Youtube-dl and yt-dlp. I would recommend you to experiment to learn how to use it a little bit better.

**Comment by nachfuellbar** [comment]
THAT is exactly what I meant.
If you need to experiment, the user experience is either not good or the documentation is.

My primary issue was - there is an "easy" solution for subtitles to filter for language and from there I searched an similar solution.
After I failed I tried many other things but didn't try the filter variation because it didn't register with me. Just used Ctrl + f to search for language and there were no examples besides something like crunchyroll etc with their extractors.

The whole README is just overloaded and seems to target user and developer together - maybe moving some info to other places and just adding a link to the other markdown file "hey here is developer info" would improve the situation.

Don't get me wrong - for me the issue is solved because I got a working solution - just thought about others in the future. If you think it's already good as is, I'll drop this ;)

**Comment by pukkandan** [comment]
You wanted to download a different format than the default, but didn't think to look at "format selection" section? If you searched for "language", you should have gotten results in the "filtering formats" and "sorting formats" sections.

> would be great if there are more examples for something like this in the readme

There are in fact examples for format selection. Yes, there is no example that is specifically for language, but there are for other fields (ext, resolution, fps, codec etc). Or do you expect examples for every use-case?

> --video-multistreams, audio-multistreams and --extractor-args "hidive:language=ja" a

And how did you get the impression these are what you want? I honestly can't see how you thought multistreams have anything to do with language. And the extractor-args you used aren't even valid. I want to try and understand what exactly in the docs confused you

> The whole README is just overloaded and seems to target user and developer together - maybe moving some info to other places and just adding a link to the other markdown file "hey here is developer info" would improve the situation.

I won't argue that the readme is a lot of info. But all the documentation except "Embedding" section is in fact for the end user, not for developers! Just because you don't find some parts useful doesn't mean it isn't for users. (The sad thing is, developer docs for yt-dlp is actually very bad/almost non-existent)

If you have any *actionable* feedback on how to improve the docs, I'm all ears

**Comment by CZeil** [comment]
@nachfuellbar That's why I was telling you, you need to read the readme. In my beginnings I used to read it almost every 10 minutes. Even now that I can use the code a little better I need to consult the readme from time to time. It's mostly that you experiment to get to use the code in a better way. We are all newbies more than once.

**Comment by nachfuellbar** [comment]
@pukkandan sorry didn't mean to offend you. Just thought others would have had similiar difficulties with using yt-dlp for a first time. 

Yes your point of view is valid - I just didn't encounter this section via a quick search.
My whole point wasn't to critize it just because I didn't find my answer. I don't exactly know which infos are necessary in this whole README - my example with the developer informations was only because it's obvious it could be separated.

My attempt with the extractor-args was only because it was with example where it was explained for e.g. crunchyroll - seemed just more apparent than the mix of "language" and "lang" in the filtering formats and sorting sections - the other reason would be because as someone who doesn't know better, i wouldn't have associated "formats" with language - if i had known this product more, it would be the other way.

To express it like a total noob - i would expect an option like "--sub-langs" for audio, too. Don't know if something like this is just not documented or if there are technical reasons for not offering something like that.

@CZeil forgot until now but thanks a lot

**Comment by pukkandan** [comment]
I didn't take any offense. I am trying to understand why you found it difficult to get to the relevant docs

> my example with the developer informations was only because it's obvious it could be separated.

It is not "obvious" though... Since like I said above, there is no developer docs mixed in. So idk what parts you are talking about

> My attempt with the extractor-args was only because it was with example where it was explained for e.g. crunchyroll

The section explicitly explains that these args are specific to each extractor, no?

> I just didn't encounter this section via a quick search.

What I want to understand is "why". The first thing I would think to search for is "language" and this will immediately bring you to "filtering formats" section. What can be added to the docs to better point you there?

>  the other reason would be because as someone who doesn't know better, i wouldn't have associated "formats" with language -

This is fair; but again, you would have seen the relation when searching for "language", no?

> To express it like a total noob - i would expect an option like "--sub-langs" for audio, too. Don't know if something like this is just not documented or if there are technical reasons for not offering something like that.

Are you suggesting there should be options for each field? like --audio-lang, --video-resolution, --video-fps, --download-protocol, etc? That doesn't make sense to me at all when the format filtering can do all these, and do them better than individual options can ever do. The --sub-lang option on the other hand is a terrible implementation. You can look at related open issues to see how limited it is. 

---

tldr; I understand your criticism that it was too hard to find this in docs. But I don't understand what you are proposing to improve it

**Comment by Jules-A** [comment]
>     * a bit confusing are the extractor args - when sorting and filtering is THE common solution, why need something in the extractors (could seem like "hey filtering doesn't work so there are extractor args which do a better job" and "maybe it's just for this extractor not explicitly documented" - you know many projects aren't exactly completely documented)

Filtering downloads everything then filters based on the results. With the Funimation extractor args it will only download webpage data for the languages you list, saving quite a lot of time on shows with many languages vs filtering.
